### PR TITLE
Added costmap in goal_checker plugins

### PR DESCRIPTION
**Pull request by hardesh**
Addresses https://github.com/ros-planning/navigation2/issues/2939 

Tested the build with ROS2 Humble

### CODE REVIEW

File: nav2_controller/include/nav2_controller/plugins/simple_goal_checker.hpp [hunk 1]
```diff
@@ -79,6 +80,8 @@ class SimpleGoalChecker : public nav2_core::GoalChecker
   rclcpp::node_interfaces::OnSetParametersCallbackHandle::SharedPtr dyn_params_handler_;
   std::string plugin_name_;
 
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros_;
```
Comment (SteveMacenski) [inline nav2_controller/include/nav2_controller/plugins/simple_goal_checker.hpp:83]: You don't need to store if its not going to be used. In the `cpp` file just change `costmap_ros` to `/*costmap_ros*/` to make the linters happy (so no arguments unused)

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
LGTM conceptually, just a couple of tiny changes.

The biggest thing that also needs updated are 
- Migration guide entry that we made this change

File: nav2_core/include/nav2_core/goal_checker.hpp [hunk 2]
```diff
@@ -68,7 +71,9 @@ class GoalChecker
    */
   virtual void initialize(
     const rclcpp_lifecycle::LifecycleNode::WeakPtr & parent,
-    const std::string & plugin_name) = 0;
+    const std::string & plugin_name,
+    const std::shared_ptr<nav2_costmap_2d::Costmap2DROS> &) = 0;
```
[the remark this answers is not in the document]
Comment (hardesh) [reply nav2_core/include/nav2_core/goal_checker.hpp:75]: ahh interesting. did not know about this

**Review by SteveMacenski — APPROVED** [review]
LGTM, please add a section to the migration guide about the API change to the goal checkers https://navigation.ros.org/migration/Galactic.html

**Comment by hardesh** [comment]
I've opened a PR in the docs repository. Missed it yesterday

**Comment by SteveMacenski** [comment]
You have a linting error but then can merge

```

-  const std::shared_ptr<nav2_costmap_2d::Costmap2DROS> /*costmap_ros*/)
+  const std::shared_ptr<nav2_costmap_2d::Costmap2DROS>/*costmap_ros*/)
```

**Pull request merged**